src/MainWindow.vala

```diff
@@ -319,17 +323,27 @@ public class Mail.MainWindow : Hdy.ApplicationWindow {
 
         unowned Mail.Backend.Session session = Mail.Backend.Session.get_default ();
 
-        session.account_removed.connect (() => {
+        var account_removed_handler = session.account_removed.connect (() => {
             var accounts_left = session.get_accounts ();
             if (accounts_left.size == 0) {
                 get_action (ACTION_COMPOSE_MESSAGE).set_enabled (false);
                 search_entry.sensitive = false;
             }
         });
 
+        var connection_handler = session.notify["online"].connect (() => {
```

Comment:
I think this value could be different per-account so I think we probably need to make sure we're updating this to reflect the online status of the currently selected mailbox. Either that or we should tie this to the network monitor and make sure that the copy reflects that by "offline" we mean disconnected from the network

Reply:
With the current setup it's basically tied to a `NetworkMonitor` with the advantage of being able to display when the `Session` is manually put to offline mode. I'm not sure, but I think for now this is the most practical way, for the reason explained in the comment above. The only edge case I could think of would be if the server of a certain provider is offline (there might be more so correct me if I'm wrong).

Reply:
I'll remove the UI part for now, as I am currently working on subclassing a `Gtk.Box` in `ConversationListBox` and will move the corresponding stuff from `MainWindow` to there. I could then implement such a banner per account in a separate PR, what would you think about that? Or should I entirely leave it away for now?